cli/azd/pkg/graphsdk/service_principal_models.go

```diff
@@ -0,0 +1,26 @@
+package graphsdk
+
+// A Microsoft Graph Service Principal entity.
+type ServicePrincipal struct {
+	Id                     *string `json:"id"`
+	DisplayName            string  `json:"displayName"`
+	AppId                  string  `json:"appId"`
+	AppOwnerOrganizationId *string `json:"appOwnerOrganizationId"`
+	AppDisplayName         *string `json:"appDisplayName"`
+	Description            *string `json:"appDescription"`
+	Type                   *string `json:"servicePrincipalType"`
+}
+
+type ServicePrincipalCreateRequest struct {
+	AppId string `json:"appId"`
+}
+
+// A list of service principals returned from the Microsoft Graph.
+type ServicePrincipalListResponse struct {
+	NextLink *string            `json:"@odata.nextLink"`
```

Comment:
Another case where we may have a pageable response?

Reply:
No needed right now.

Reply:
Added #974 to track this.